src/hash_ring.cc

```diff
@@ -37,7 +37,9 @@ HashRing::HashRing(std::map<std::string, uint32_t> weights, uint32_t precision)
   // Construct the bucket list based on the weight hash
   for (const auto& weight: weights) {
     NodeInfo *node = &(node_list[i]);
-    node->id = weight.first;
+    char* name = new char[weight.first.length()];
+    strcpy(name, weight.first.c_str());
+    node->id = name;
```

Comment:
I'm trying to understand why you need to allocate a c-style string and copy the value before assigning rather than just assigning `node->id = weight.first.c_str();`

Reply:
Yeah I think `c_str()` just gives you a `char*` that points at the memory which was allocated and managed by javascript. When we `free` the struct, we delete the memory and the later js tries to reclaim it and blows up when it has already been freed.

Reply:
Yup, that makes sense